### PR TITLE
Fix writing a loder instructions

### PR DESCRIPTION
**Pull request by danielhusar**
When I followed the tutorial for writing loader, I have noticed couple of issues, see the inline comments.

### CODE REVIEW

File: src/content/contribute/writing-a-loader.md [hunk 1]
```diff
@@ -333,15 +333,15 @@ export default (fixture, options = {}) => {
...
-          }
+          options,
         }
       }]
     }
   });
 
-  compiler.outputFileSystem = createFsFromVolume(new Volume());
+  const volume = createFsFromVolume(new Volume());
+  volume.join = path.join;
```
Comment (danielhusar) [inline src/content/contribute/writing-a-loader.md:343]: https://github.com/streamich/memfs/issues/404
https://github.com/webpack/memory-fs/issues/67

File: src/content/contribute/writing-a-loader.md [hunk 2]
```diff
@@ -377,6 +377,9 @@ __package.json__
 {
   "scripts": {
     "test": "jest"
+  },
+  "jest": {
+    "testEnvironment": "node"
```
Comment (danielhusar) [inline src/content/contribute/writing-a-loader.md:382]: https://github.com/facebook/jest/issues/1909#issuecomment-252981401

File: src/content/contribute/writing-a-loader.md [hunk 3]
```diff
@@ -333,15 +333,15 @@ export default (fixture, options = {}) => {
         test: /\.txt$/,
         use: {
           loader: path.resolve(__dirname, '../src/loader.js'),
-          options: {
-            name: 'Alice'
-          }
+          options,
```
Comment (danielhusar) [inline src/content/contribute/writing-a-loader.md:336]: I think it's bit nicer to pass options from your tests so this is more reusable.

File: src/content/contribute/writing-a-loader.md (hunk 1 above)
Comment (danielhusar) [reply src/content/contribute/writing-a-loader.md:343]: Without this, it throws error: `TypeError: this.outputFileSystem.join is not a function`

**Comment by chenxsan** [comment]
Oh, I didn't notice there's already a similar pull request here before I submit https://github.com/webpack/webpack.js.org/pull/3739. I should search the pull request first next time.

**Review by EugeneHlushko — APPROVED** [review]

**Comment by EugeneHlushko** [comment]
Thanks!

**Pull request merged**